### PR TITLE
Fix linker error caused by VLOG macro

### PR DESCRIPTION
**Pull request by 4c3y**
If VLOG() is used in MODE_ROSLOG or MODE_LPP and glog is installed but not linked to lpp, a linker error is generated. This happens because VLOG() uses the FLAGS_v global variable and VLOG_IS_ON() macro. This PR replaces both of them with an lpp implementation if MODE_LPP or MODE_ROSLOG is used.

### CODE REVIEW

File: include/log++.h [hunk 1]
```diff
@@ -352,11 +352,19 @@ LPP_INTL::InternalPolicyLog(LPP_GET_KEY(), n, LPP_INTL::BaseSeverity::DEBUG, LPP
...
 #define LOG_STRING(severity, ptr) LPP_ASSERT_GLOG(LPP_INTL::GlogSeverity::severity); LPP_INTL::InternalGlogLogStringLog(toBase(LPP_INTL::GlogSeverity::severity), ptr)
 
-#ifndef GLOG_SUPPORTED
-inline static int32_t FLAGS_v;
-#define VLOG_IS_ON(verboselevel) FLAGS_v >= (verboselevel) ? true : false
+/**
+ * Replace glog's FLAGS_v and VLOG_IS_ON to avoid linker errors
+ * if glog is installed but not linked to lpp.
+ */
+inline static int32_t LPP_FLAGS_v;
```
Comment (rikba) [inline include/log++.h:359]: I'm getting the warning
```
/home/rik/pq_ws/src/lpp/include/log++.h:359:23: warning: ‘LPP_FLAGS_v’ defined but not used [-Wunused-variable]
  359 | inline static int32_t LPP_FLAGS_v;
```

Do you think we can do something about it?

Comment (4c3y) [reply include/log++.h]: Yes, I added the [[maybe_unused]] C++ attribute

**Comment by rikba** [comment]
When building and running the demo node (or any other executable, that does not explicitly link against rosconsole) I'm getting the following error:

```
target_compile_definitions(lpp_demo PRIVATE MODE_ROSLOG)
```

```
./lpp_demo: error while loading shared libraries: librosconsole_log4cxx.so: cannot open shared object file: No such file or directory
```

I find this weird, because the demo node does link against roscpp (https://github.com/ethz-asl/lpp/blob/master/CMakeLists.txt#L49). Any idea @4c3y

**Comment by rikba** [comment]
> When building and running the demo node (or any other executable, that does not explicitly link against rosconsole) I'm getting the following error:
> 
> ```
> target_compile_definitions(lpp_demo PRIVATE MODE_ROSLOG)
> ```
> 
> ```
> ./lpp_demo: error while loading shared libraries: librosconsole_log4cxx.so: cannot open shared object file: No such file or directory
> ```
> 
> I find this weird, because the demo node does link against roscpp (https://github.com/ethz-asl/lpp/blob/master/CMakeLists.txt#L49). Any idea @4c3y

Sorry, ignore my comment. I did not source ROS....

**Review by rikba — APPROVED** [review]
LGTM! Thanks

**Pull request merged**